assets/js/components/notifications/FirstPartyModeSetupBanner.js

```diff
@@ -67,21 +67,22 @@ export default function FirstPartyModeSetupBanner( { id, Notification } ) {
 		select( CORE_NOTIFICATIONS ).isNotificationDismissed( id )
 	);
 
-	const { dismissNotification, invalidateResolution } =
-		useDispatch( CORE_NOTIFICATIONS );
+	const { invalidateResolution } = useDispatch( CORE_NOTIFICATIONS );
 	const { setValue } = useDispatch( CORE_UI );
 
-	const onCTAClick = () => {
+	const onCTAClick = async () => {
 		setFirstPartyModeEnabled( true );
-		saveFirstPartyModeSettings();
+		const { error } = await saveFirstPartyModeSettings();
+
+		if ( error ) {
+			return { error };
+		}
 
 		setValue( FPM_SHOW_SETUP_SUCCESS_NOTIFICATION, true );
 		invalidateResolution( 'getQueuedNotifications', [
 			viewContext,
 			NOTIFICATION_GROUPS.DEFAULT,
 		] );
-
-		dismissNotification( id );
```

Comment:
This is a cleanup that is unrelated to the error handling.

We can avoid dismissing the notification here, as we can fully dismiss it in the `CTALink` component:

https://github.com/google/site-kit-wp/blob/a19d37db70b1d78525732e0585354d8032fa19d1/assets/js/googlesitekit/notifications/components/common/CTALink.js#L93-L100

This avoids calling the `dismiss-item` endpoint twice when dismissing the component.

Reply:
Thanks @techanvil for the cleanup here - I saw this and was going to remove the additional call in another PR. We also need to remove the `isItemDismissed` logic.

Reply:
Thanks @jimmymadon. However - it looks like we need to keep the `isItemDismissed` logic as things stand, because the notification won't automatically disappear without it.

This is due to the use of the `skipHidingFromQueue` dismiss option below, which in turn is needed to allow the tooltip to be displayed via this component.

Reply:
Ahh we should probably write some inline comments here, with a Todo to revisit this holistically to either separate out the notification and the tooltip components if possible. Or we now also have actions that allow us to remove and add items to the notification queue which might help here. But I'll look at this again when dealing with a related notifications issue Im working on.

Reply:
Sounds good. I'll leave this for the developer who executes the issue to follow up on.